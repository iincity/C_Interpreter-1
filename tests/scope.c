#include <stdio.h>


int a = 10 % 3, b = 13 * 12 + 2, c = 10;

int main()
{
	printf("%d\n", a);
	printf("%d\n", b);
	printf("%d\n", c);

	int d = 12;
	{
		printf("%d\n", d);
		printf("%d\n", a);
		int a = 15;
		printf("%d\n", a);
		{
			printf("%d\n", d);
			printf("%d\n", a);
			printf("%d\n", b);

			int b = 100;
			printf("%d\n", b);
			printf("%d\n", c);
			{
				printf("%d\n", c);
			}

		}

		{
			printf("%d\n", b);
			int b = 50;
			printf("%d\n", b);
		}

		int c = 5;
		printf("%d\n", c);

	}


	return 0;
}


