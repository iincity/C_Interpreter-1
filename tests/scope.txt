int a = 10 % 3, b = 13 * 12 + 2, c = 10;

int main()
{
	print a;
	print b;
	print c;

	int d = 12;
	{
		print d;
		print a;
		int a = 15;
		print a;
		{
			print d;
			print a;
			print b;

			int b = 100;
			print b;
			print c;
			{
				print c;
			}

		}
		int c = 5;
		print c;
	}


	return 0;
}


