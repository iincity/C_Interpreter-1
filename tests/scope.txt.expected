1
158
10
12
1
15
12
15
158
100
10
10
158
50
5
